### DISCUSSION TITLE
How to add a new classification to the training result file?

### DISCUSSION THREAD

**Discussion by m2msoft** [ultralytics/ultralytics#1095] [Q&A]
I just finished trarning one classification with YoloV8.

I want to add a new classification here.  But, when setting the existing learning results as a model and trarning a new classification, the existing trarning results are ignored.
Should the training data be reclassified and fully regenerated? Or, is there a way to add new results to existing results?

Thank you

`yolo train model=best_20230222.pt data=./data/data.yaml epochs=10 imgsz=640 device=0 batch=32 workers=4`

**Comment by glenn-jocher** [comment]
@m2msoft hello! It's great to hear that you've successfully trained a classification model using YOLOv8. If you're looking to add a new class to your model, you'll need to update your dataset to include examples of the new class and then continue training (also known as fine-tuning) your model with the updated dataset.

Here's a general outline of the steps you should follow:

1. Update your dataset to include the new class, ensuring you have labeled images for this class.
2. Modify your `data.yaml` file to include the new class in the list of classes.
3. Load your previously trained model weights using the `model` argument.
4. Continue training with the updated dataset and `data.yaml` file.

By following these steps, your model should learn to recognize the new class while also retaining its ability to recognize the previously trained classes. Remember to monitor the validation metrics to ensure that your model is not forgetting the previous classes, a phenomenon known as "catastrophic forgetting."

For more detailed instructions on training, you can refer to the training documentation at https://docs.ultralytics.com/modes/train. Good luck with your model enhancement! 🚀
